rust/feature-flags/src/flag_matching.rs

```diff
@@ -805,6 +824,85 @@ impl FeatureFlagMatcher {
         }
     }
 
+    pub async fn evaluate_cohort_filters(
+        &self,
+        filters: &[PropertyFilter],
+        target_properties: &HashMap<String, Value>,
+    ) -> Result<bool, FlagError> {
+        Box::pin(self.evaluate_potentially_nested_cohort_filters(filters, target_properties)).await
+    }
+
+    async fn evaluate_potentially_nested_cohort_filters(
+        &self,
+        filters: &[PropertyFilter],
+        target_properties: &HashMap<String, Value>,
+    ) -> Result<bool, FlagError> {
+        let mut cohort_filters = Vec::new();
+        let mut non_cohort_filters = Vec::new();
+
+        // Separate cohort filters from non-cohort filters
+        for filter in filters {
+            if filter.prop_type == "cohort" {
+                cohort_filters.push(filter);
+            } else {
+                non_cohort_filters.push(filter);
+            }
+        }
```

Comment:
It's not so much that I'm _separating_, I'm _differentiating_.  Basically, non_cohort_filters can be evaluated using `match_property`, but `cohort_filters` have to be pulled apart until they are property filters.  This lets us evaluate cohorts properties like this
```
{"properties": {"type": "OR", "values": [{"type": "OR", "values": [{"key": "id", "type": "cohort", "value": 8, "negation": false}, {"key": "$browser", "type": "person", "value": ["Safari"], "negation": false, "operator": "exact"}]}]}}
```

where one property is a regular person property, and the other one needs to look up the cohort to evaluate.